UnityProject/Assets/Scripts/Items/Tool/Soap.cs

```diff
@@ -33,6 +33,24 @@ public bool WillInteract(PositionalHandApply interaction, NetworkSide side)
 			//can only scrub tiles, for now
 			if (!Validations.HasComponent<InteractableTiles>(interaction.TargetObject)) return false;
 
+            // Get position of interaction as Vector3Int
+            Vector3 position = interaction.WorldPositionTarget;
+            Vector3Int positionInt = Vector3Int.RoundToInt(position);
+
+            // Check if there is an object in the way of scrubbing the tile
+			var atPosition = MatrixManager.GetAt<RegisterTile>(positionInt, true);
```

Comment:
This will block if theres items on the floor, might be better to use `RegisterObject` instead?